### DISCUSSION TITLE
How to handle activation of Toast Notifications on Windows?

### DISCUSSION THREAD

**Discussion by emorell96** [dotnet/maui#5945] [Q&A]
Hi,

I was working on handling activation of toast notifications using the UWP Toast Toolkit (https://docs.microsoft.com/en-us/windows/apps/design/shell/tiles-and-notifications/send-local-toast?tabs=uwp) and I was wondering if I could hook up to the OnActivated event of UWP apps since MAUI uses UWP for its windows implementation.

On UWP usually you just hook up to this event on App.xaml.cs:
![image](https://user-images.githubusercontent.com/20151415/162576588-91e49f30-d144-4613-9afd-135fdfe0114d.png)

But that event is not the  same as the one available on MAUI AFAIK.

Is it possible to handle the activation of the toast notification? Or do I have to follow the Desktop MSIX instructions of that page?

Thank you!

**Comment by redwolf2** [comment]
https://github.com/gurrenm3/Toast-Notifications-with-.Net-Maui

**Reply by flaubertagufonku** [reply]
This is the type of tutorial everyone needs. A full example with the particularities of WinUI which removed completly the behavior of opening a new app when the user clicks on a notification. At the same time, it allows us to handle the notification prompted.
Nice job and thank you.

**Comment by mhrastegari** [comment]
You can use .NET MAUI Community Toolkit for that
https://learn.microsoft.com/en-us/dotnet/communitytoolkit/maui/alerts/toast

Don't miss guided by the Android and iOS screenshots the Windows implementation is exactly what you need!
